htdocs/install/mysql/migration/18.0.0-19.0.0.sql

```diff
@@ -120,5 +120,14 @@ ALTER TABLE llx_accounting_account MODIFY COLUMN pcg_type varchar(32);
 -- VPGSQL8.2 DROP INDEX uk_links;
 ALTER TABLE llx_links ADD UNIQUE INDEX uk_links (objectid, objecttype,label);
 
+ALTER TABLE llx_facture MODIFY COLUMN subtype varchar(5) DEFAULT NULL;
```

Comment:
There is a pb with that. The table of subtype has a unique id that is the field rowid. It seems you plan to link the facture to table of type of invoice with field code. That is not unique. 2 entities can have the same code or example 1 company in france and another one in argentina. With such a link, you will get a duplicate.
Foreign key must be on the rowid so subtype must be an int field.

Reply:
Hi @eldy
> It seems you plan to link the facture to table of type of invoice with field code. That is not unique. 2 entities can have the same code or example 1 company in france and another one in argentina. With such a link, you will get a duplicate.

You are right but i chose to go this way because of classification api where
 | field | Type | 
| ---- | ---|
| invoiceType | xs:string |

and your example use case  is handled in html.form.class.php and function load_cache_invoice_subtype() 
```

		$sql = "SELECT rowid, code, label as label";
		$sql .= " FROM " . MAIN_DB_PREFIX . 'c_invoice_subtype';
		$sql .= " WHERE entity IN (" . getEntity('c_invoice_subtype') . ")";
		$sql .= " AND active = 1";
```

Is not this enough since the user will have to disable other countries entries from dictionary anyway?


![dict](https://github.com/Dolibarr/dolibarr/assets/93765174/e871702e-3b47-4a98-8f07-56185beda0ae)

Your remark though led me to discover this in llx_c_invoice_subtype.key.sql

`ALTER TABLE llx_c_invoice_subtype ADD UNIQUE INDEX uk_c_invoice_subtype (entity, code);`

Should that be changed to 

`ALTER TABLE llx_c_invoice_subtype ADD INDEX uk_c_invoice_subtype (entity, code);` ?

Thank you for your patience, please advice

Reply:
> Is not this enough since the user will have to disable other countries entries from dictionary anyway?

To answer my own question, no it's not